grammars/level11-Additions.lark

```diff
@@ -1,11 +1,13 @@
-command:+= for_loop | call | define
+command:+= for_loop | call | error_nested_define
 
 //new : for loop
 for_loop: _FOR var _IN _RANGE (INT | var_access) _TO (INT | var_access) _EOL (_SPACE command) (_EOL _SPACE command)*  _EOL? _END_BLOCK
 
 // FUNCTIONS ======================================== =============================
 // check line 1: command:+= call | define
+_non_empty_program: _EOL* (define | command | error_invalid) _SPACE* (_EOL+ (define | command) _SPACE*)* _EOL* //lines may end on spaces and might be separated by many newlines
```

Comment:
Thanks for the addition @jlole!

I don't love the fact that define is a bit hidden here, it is a command, could we not merge it with command in line 1? I guess there is a reason for this (a conflict in parsing with another command, because you put it before) but we can redefine `command` here, like in level 6, so it is more clear where the conflict is.

Reply:
The reason for this is that if `define` is merged with `command`, I cannot check if a function definition is nested. This way, `define` is only allowed outside of any blocks (can't define a function in another function or in an if statement)